### PR TITLE
Add Circle CI config

### PR DESCRIPTION
**Pull request by morrisonlevi**
This allows us to run certain jobs publicly instead of them all being
hidden away in GitLab.

I plan to add some .phpt tests as a follow-up PR.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+      - run:
+          name: Add OS packages
+          command: |
+            set -euo pipefail
+            sudo apt-get update
+            sudo apt-get install -y php7.4-dev libuv1-dev
+
+      - run:
+          name: Add /opt packages
```
Comment (r1viollet) [inline .circleci/config.yml:24]: circle CI has a pretty cool caching mechanism, we could use concatenation of both keys as a cache for the job and indicate /opt as result of the cache.

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            sudo cp -v cmake-build-release/datadog-profiling.so "$extension_dir"
+            ini_dir="$(php-config --ini-dir)"
+            sudo mkdir -vp "$ini_dir"
+            echo "zend_extension=datadog-profiling.so" > datadog-profiling.ini
+            sudo cp -v datadog-profiling.ini "$ini_dir/"
+
+      - run:
+          name: Run php --ri diagnostics
+          command: |
+            # This should "fail" with a success error code with a message
```
Comment (r1viollet) [inline .circleci/config.yml:70]: Should we do the `set -euo pipefail` here ?

**Review by r1viollet — APPROVED** [review]
Thanks for setting up the initial CI structure !
LGTM

File: .circleci/config.yml (hunk 1 above)
Comment (morrisonlevi) [reply .circleci/config.yml:24]: Great idea. Maybe we can cache in another PR.

**Pull request merged**